src/web/components/StickyBottomBanner/BrazeBanner.tsx

```diff
@@ -114,6 +131,20 @@ const getMessageFromBraze = async (
         appboy.changeUser(brazeUuid);
         appboy.openSession();
     });
+
+    canShowPromise.then(() => {
+        const appboyTimeTaken = appboyTiming.end();
+
+        record({
+            component: 'braze-appboy-timing',
+            value: appboyTimeTaken,
+        });
+    }).catch(() => {
+        // eslint-disable-next-line no-console
+        console.log("Appboy Timing failed.");
+    });
```

Comment:
this is more of a question, but would there be any benefit to running `const appboyTimeTaken = appboyTiming.end();` here?

Reply:
Thanks for looking over this! My feeling is we would not want to log a timing value if the promise results in an error - though we could handle the error more elegantly than we do here.

What do you reckon @tjmw ?

Reply:
I thought the logging was done by `record`? I was snooping around on MDN and found this https://developer.mozilla.org/en-US/docs/Web/API/Performance/clearMarks   Might be worth creating another export in `initPerf` like:

```
const close = () => {
        perf. clearMarks(startKey);
    };
```

we could then close it by running `appboyTiming .close()`

Reply:
You are right, I had assumed you were suggesting we `.end()` and `record` since there would not be much value in running `.end()` by itself given that it will add a second performance mark and measure the difference between the two without using the value to do anything.

We could clear the mark - but I don't know if the performance mark sitting in memory somewhere will have a greater impact on performance than any attempt we could make to remove/close it e.g. via `clearMarks`. My assumption is that that they don't persist beyond the page (and are discarded), but I don't know for sure, nor can I find the answer in the documentation.

Reply:
I think if you have started a performance mark, and are not going to end it, it makes sense to close it in the catch block. I doubt this really has any major performance benefit, but I guess its just good practice to stop unnecessary background calculations.

Reply:
Hey I agree with @liywjl that it does _feel_ cleaner to clear the mark in the `catch` block, though also agree that there's probably not a significant performance benefit to doing so (I'm not too sure how the performance API is implemented under the hood - but presumably not clearing would just mean an unused mark entry stored somewhere?) There's probably also an argument that clearing the mark makes the intent clearer to future developers reading the code. I reckon let's go ahead an add this since it seems like a pretty straightforward change.

Reply:
Cool, I've added a .clear() method along the lines that @liywjl suggested, which we can call to tidy up marks when we don't call .end() - i.e. when we catch an error.

Reply:
Thanks @rhystmills I just noticed you have a lint issue because `// eslint-disable-next-line no-console` should be one line down. Bar that all looks good to merge

Reply:
So it is, thanks for catching that.